pkg/custmsg/custom_message.go

```diff
@@ -11,16 +11,41 @@ import (
 	"github.com/smartcontractkit/chainlink-common/pkg/values"
 )
 
-type Labeler struct {
+type Labeler interface {
+	WithMapLabels(labels map[string]string) Labeler
+	With(keyValues ...string) Labeler
+	Emit(msg string) error
+	Labels() map[string]string
+}
+
+type labeler struct {
 	labels map[string]string
 }
 
-func NewLabeler() Labeler {
-	return Labeler{labels: make(map[string]string)}
+func NewLabeler() *labeler {
+	return &labeler{labels: make(map[string]string)}
+}
+
+// WithMapLabels adds multiple key-value pairs to the CustomMessageLabeler for transmission
+// With SendLogAsCustomMessage
+func (c *labeler) WithMapLabels(labels map[string]string) *labeler {
+	newCustomMessageLabeler := NewLabeler()
+
+	// Copy existing labels from the current agent
+	for k, v := range c.labels {
+		newCustomMessageLabeler.labels[k] = v
+	}
+
+	// Add new key-value pairs
+	for k, v := range labels {
+		newCustomMessageLabeler.labels[k] = v
+	}
+
+	return newCustomMessageLabeler
 }
 
 // With adds multiple key-value pairs to the CustomMessageLabeler for transmission With SendLogAsCustomMessage
-func (c Labeler) With(keyValues ...string) Labeler {
+func (c *labeler) With(keyValues ...string) *labeler {
```

Comment:
/nit all the methods on Labeler are intended to be read-only, like with Logger, so I do not think pointer receivers are necessary